fl4health/model_bases/autoencoders_base.py

```diff
@@ -0,0 +1,144 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Tuple
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class AutoEncoderType(Enum):
+    BASIC_AE = "BASIC_AE"
+    VARIATIONAL_AE = "VARIATIONAL_AE"
+    CONDITIONAL_VAE = "CONDITIONAL_VAE"
+
+
+class AutoEncoderBase(nn.Module, ABC):
```

Comment:
To avoid confusion with Basic_AE, it might be best to call this AbstractAE perhaps?

Reply:
Definitely a better name!